### PR TITLE
Extinguisher + Container Reworks

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Changes:

- Reworked how slipping on tiles work. If there is too much lube/water on one tile, it will spread to all adjacent tiles.
- Fixed a potential bug where spilling copious amounts of water on a tile will turn it to ice.

- Containers are much more saner now. They all share a standard system, unless specified.
- You can drink from any open holdable container using help intent. You can splash the contents of the container on anything via harm intent.
- Sinks are more saner, you can refill or empty them easier with various refill amounts.
- Removed silly exclusive cyborg limb checking for hyposprays.
- Added a new chemsprayer, the "Xenoblaster" which is a significantly weaker chem sprayer. Two can be found in xenobiology.
- Large reagent containers (beer kegs, water tanks, fuel tanks) now share a system together. Both can leak their reagents via harm intent with a wrench, and bother can accept any sort of reagent, with exception to the fuel tank.
- Large reagent container leaking happens over time as opposed to just leaking every time it moves.
- New chemical, Monoammonium phosphate. Monoammonium phosphate is a fertilizer and also a potent fire extinguishing and preventing chemical. It can be made with 1 part ammonia, 1 part sulfuric acid, 1 part sodium, 1 part phosphorus.
- All extinguishers now have monoammonium phosphate in them instead of water.
- Extinguishers can hold any reagent, however they must be filled using an extinguisher refiller cartridge. These can be ordered or found in atmospherics.
- Nerfed the extinguisher range so it only sprays in a 3x3 radius.
- Mini-extinguishers spray in a 1x1 radius.
- Adds Monoammonium phosphate containers across the station.
- Most watertanks are replaced with Monoammonium phosphate containers.

### CODE REVIEW

**Comment by BotBOREALIS** [comment]
Icons with diff:
<details><summary>icons/obj/objects.dmi</summary>

Key | Old | New | Status
--- | --- | --- | ---
extinguisher_tank||extinguisher_tank|Created
</details>

**Comment by Arrow768** [comment]
Please post a screenshot of the mapping changes

**Comment by BurgerLUA** [comment]
arrow is into skinny bitches and told me to remap it so it isn't massive, this is the new design

yes

**Comment by LordFowl** [comment]
Have you tested these mapping changes ingame? I am pretty sure the corner fire extinguisher tank is inaccessible, as well as the two corner tables.

**Comment by Alberyk** [comment]
Is this moving the atmos' axe? If so, why does atmos need an extra axe?

**Comment by BurgerLUA** [comment]
I know the fire extinguisher tank in the corner cannot be accessed. It's there as an extra tank.

If it really bothers you, I can remove the axe in the lobby. so there is a fire axe for each atmos tech.

**Comment by Alberyk** [comment]
I believe that a single axe for atmos is more than enough in any case, it is not really an essential piece of gear like voidsuits.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 1]
```diff
@@ -264,6 +264,46 @@
...
+	taste_description = "salty dirt"
+	metabolism = REM * 2
+
+/datum/reagent/toxin/fertilizer/monoammoniumphosphate/touch_turf(var/turf/simulated/T)
+	if(!istype(T))
+		return
+
+	var/hotspot = (locate(/obj/fire) in T)
+	if(hotspot && !istype(T, /turf/space))
+		var/datum/gas_mixture/lowertemp = T.remove_air(T:air:total_moles)
```
Comment (skull132) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: Bad use of colon access operators. T is known as a turf. Just replace with periods.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 2]
```diff
@@ -264,6 +264,46 @@
...
+
+/datum/reagent/toxin/fertilizer/monoammoniumphosphate/touch_mob(var/mob/living/L, var/amount)
+	if(istype(L))
+		L.adjust_fire_stacks(-amount)
+		if(L.fire_stacks <= 0 )
+			L.fire_stacks = 0
+			L.ExtinguishMob()
+
+/datum/reagent/toxin/fertilizer/monoammoniumphosphate/affect_touch(var/mob/living/carbon/M, var/alien, var/removed)
+	if(istype(M, /mob/living/carbon/slime))
```
Comment (skull132) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: Cast parameter as slime and do `if (istype(S))`. No need for the extra typecast down below.

**Comment by Alberyk** [comment]
It would also make chemsprayers pointless, since you can get them anywhere, I am also against adding this feature to fire extinguishers.

**Comment by BurgerLUA** [comment]
Arrow told me there was a discussion on this so I'm actively improving this PR.

Here's what's coming:
Extinguishers can only be refilled by people with extinguisher refiller tanks. These will spawn only in engineering, but of course they will be orderable. It works like inhalers where you put in the reagents, twist the cap, and then use it on an extinguisher to refill it. The cartridge cannot be reused however one cartridge can be used to fill multiple tanks, if there are reagents available. I personally don't like this but this is the only way to address your guy's complaints. Perhaps I can make an extinguisher refilling station that only refills extinguishers with the fluid required.

Xenobiology will get it's own custom chem sprayer instead of fire extinguishers. It's a slightly better spray bottle, worse than a chem sprayer, but it can be filled with anything.

File: code/modules/reagents/reagent_containers/borghydro.dm [hunk 3]
```diff
@@ -51,8 +51,14 @@
 					reagent_volumes[T] = min(reagent_volumes[T] + 5, volume)
 	return 1
 
-/obj/item/weapon/reagent_containers/borghypo/attack(var/mob/living/M, var/mob/user)
+/obj/item/weapon/reagent_containers/borghypo/afterattack(var/obj/target, var/mob/user, proximity)
+
+	if(!proximity)
+		return
+
+	var/mob/living/M = target
```
Comment (skull132) [inline code/modules/reagents/reagent_containers/borghydro.dm]: Why this change? The initial argument was fine.

File: code/modules/reagents/reagent_containers/hypospray.dm [hunk 4]
```diff
@@ -20,11 +20,18 @@
...
 
-/obj/item/weapon/reagent_containers/hypospray/attack(mob/living/M as mob, mob/user as mob)
-	if(!reagents.total_volume)
-		user << "<span class='warning'>[src] is empty.</span>"
+/obj/item/weapon/reagent_containers/hypospray/afterattack(obj/target, mob/user, proximity)
+
+	if(!proximity)
 		return
+
+	var/mob/living/M = target
```
Comment (skull132) [inline code/modules/reagents/reagent_containers/hypospray.dm]: :ree: junk variables which could be avoided with argument casting.

File: code/modules/reagents/reagent_containers/inhaler.dm [hunk 5]
```diff
@@ -14,57 +14,59 @@
...
 
-/obj/item/weapon/reagent_containers/inhaler/attack(mob/living/M as mob, mob/user as mob)
+/obj/item/weapon/reagent_containers/inhaler/afterattack(var/obj/target, var/mob/user, proximity)
 
-	if(!reagents.total_volume)
-		to_chat(user,"<span class='warning'>\The [src] is empty.</span>")
-		return
 
-	var/mob/living/carbon/human/H = M
+	var/mob/living/carbon/human/H = target
```
Comment (skull132) [inline code/modules/reagents/reagent_containers/inhaler.dm]: y u do dis.

File: code/modules/reagents/reagent_containers/spray.dm [hunk 6]
```diff
@@ -26,13 +26,19 @@
...
 
 
 /obj/item/weapon/reagent_containers/spray/afterattack(atom/A as mob|obj, mob/user as mob, proximity)
-	if(istype(A, /obj/item/weapon/storage) || istype(A, /obj/structure/table) || istype(A, /obj/structure/closet) || istype(A, /obj/item/weapon/reagent_containers) || istype(A, /obj/structure/sink) || istype(A, /obj/structure/janitorialcart))
+
+	if(istype(A, /obj/item/weapon/reagent_containers))
+		. = ..()
 		return
 
+	for(var/type in can_be_placed_into)
```
Comment (skull132) [inline code/modules/reagents/reagent_containers/spray.dm]: Psure we have a helper for this? `is_type_in_list` or something?

File: code/modules/reagents/reagent_dispenser.dm [hunk 7]
```diff
@@ -58,15 +55,34 @@
...
+				return
 
-
-
-
-
+			var/obj/item/weapon/reagent_containers/RG = O
+			if (istype(RG) && RG.is_open_container())
+				var/atype = alert(usr, "Do you want to fill or empty \the [RG] at \the [src]?", "Fill or Empty", "Fill", "Empty", "Cancel")
+				if(!usr.Adjacent(src)) return
```
Comment (skull132) [inline code/modules/reagents/reagent_dispenser.dm]: `usr` in sublevel code. Swap to `user`.

File: code/modules/reagents/reagent_dispenser.dm [hunk 8]
```diff
@@ -58,15 +55,34 @@
...
+				if(RG.loc != usr) return
```
Comment (skull132) [inline code/modules/reagents/reagent_dispenser.dm]: `usr` => `user`

File: code/modules/reagents/reagent_containers/borghydro.dm (hunk 3 above)
Comment (BurgerLUA) [reply code/modules/reagents/reagent_containers/borghydro.dm]: Question, if afterattack target was casted to a mob type, would it run at all if the target is actually a non-mob? I want it to run the parent afterattack if the target is not a mob.

File: code/modules/reagents/reagent_containers/spray.dm (hunk 6 above)
Comment (BurgerLUA) [reply code/modules/reagents/reagent_containers/spray.dm]: I was copying old code.

File: code/modules/reagents/reagent_dispenser.dm [hunk 9]
```diff
@@ -126,10 +126,10 @@
 
 /obj/structure/reagent_dispensers/fueltank/attack_hand()
 	if (rig)
-		usr.visible_message("[usr] begins to detach [rig] from \the [src].", "You begin to detach [rig] from \the [src]")
-		if(do_after(usr, 20))
-			usr.visible_message("<span class='notice'>[usr] detaches [rig] from \the [src].</span>", "<span class='notice'>You detach [rig] from \the [src]</span>")
-			rig.loc = get_turf(usr)
+		user.visible_message("[user] begins to detach [rig] from \the [src].", "You begin to detach [rig] from \the [src]")
```
Comment (skull132) [inline code/modules/reagents/reagent_dispenser.dm]: fuck this bug again. `user` is not defined in this scope, despite the compiler not complaining. Add `mob/user` to the proc arguments.

**Comment by BurgerLUA** [comment]
Will be putting this one in the July Update instead.

**Comment by BurgerLUA** [comment]
Todo: Fix hypospray.
Todo: Test everything else.

File: code/modules/reagents/reagent_dispenser.dm (hunk 9 above)
Comment (BurgerLUA) [reply code/modules/reagents/reagent_dispenser.dm]: fixed

**Comment by BurgerLUA** [comment]
Alright. new updates:

- Almost all tanks can leak reagents via harm intent with a wrench.
- Leaking tanks no longer relies on movement to leak reagents.
- Lube/Water now spreads to other tiles if there is too much water on one tile.

**Comment by BurgerLUA** [comment]
It's done. I've tested this extensively, however knowing my luck, there may be some potential code issues or bugs. Nothing that would crash the server or cause chaos.

!review

**Comment by BurgerLUA** [comment]
!review

File: code/game/objects/items/weapons/tools.dm [hunk 10]
```diff
@@ -338,7 +338,7 @@
 			message_admins("[key_name_admin(user)] triggered a fueltank explosion.")
 			log_game("[key_name(user)] triggered a fueltank explosion with a welding tool.",ckey=key_name(user))
 			user << span("alert", "That was stupid of you.")
-			tank.explode()
+			tank.ex_act()
```
Comment (Lohikar) [inline code/game/objects/items/weapons/tools.dm]: ex_act expects an argument.

File: code/game/turfs/simulated.dm [hunk 11]
```diff
@@ -18,26 +19,64 @@
...
+
+	//Wet type:
+	//WET_TYPE_WATER = water
+	//WET_TYPE_LUBE = lube
+	//WET_TYPE_ICE = ice
+	
+	if(!wet_type)
+		wet_type = type
+	else if(type != wet_type)
+		if(type == WET_TYPE_WATER) //Water is being added
```
Comment (Lohikar) [inline code/game/turfs/simulated.dm]: This could be a switch, and I suspect there's a more concise mathematical way to do this.

File: code/game/turfs/simulated.dm [hunk 12]
```diff
@@ -18,26 +19,64 @@
 
 	roof_type = /turf/simulated/floor/airless/ceiling
 
-/turf/simulated/proc/wet_floor(var/wet_val = 1)
-	if(wet_val < wet)
-		return
-
-	if(!wet)
-		wet = wet_val
+/turf/simulated/proc/wet_floor(var/type = WET_TYPE_WATER, var/amount = 1)
```
Comment (Lohikar) [inline code/game/turfs/simulated.dm]: `type` is a `/datum` var; masking it with a param could be a source of confusion.

File: code/modules/assembly/igniter.dm [hunk 13]
```diff
@@ -22,8 +22,8 @@
 			if (istype(src.loc.loc, /obj/structure/reagent_dispensers/fueltank/))
 				var/obj/structure/reagent_dispensers/fueltank/tank = src.loc.loc
 				if (tank && tank.modded)
-					tank.explode()
-		
+					tank.ex_act()
```
Comment (Lohikar) [inline code/modules/assembly/igniter.dm]: Once more, ex_act expects an argument.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 14]
```diff
@@ -259,6 +259,53 @@
...
+	metabolism = REM * 2
+
+/datum/reagent/toxin/fertilizer/monoammoniumphosphate/touch_turf(var/turf/simulated/T)
+
+	if(!istype(T))
+		return
+
+	var/hotspot = (locate(/obj/fire) in T)
+	if(hotspot && !istype(T, /turf/space))
+		var/datum/gas_mixture/lowertemp = T.remove_air(T.air.total_moles)
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: `air` is a ZAS-internal var that should not be touched outside of ZAS. There's a proc to return the air for an atom, use that.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 15]
```diff
@@ -259,6 +259,53 @@
...
+		L.adjust_fire_stacks(-amount)
+		if(L.fire_stacks <= 0 )
+			L.fire_stacks = 0
+			L.ExtinguishMob()
+
+/datum/reagent/toxin/fertilizer/monoammoniumphosphate/affect_touch(var/mob/living/carbon/slime/S, var/alien, var/removed)
+	if(istype(S))
+		S.adjustToxLoss(8 * removed)
+		if(!S.client)
+			if(S.Target) // Like cats
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: This could be combined into the `if` above.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 16]
```diff
@@ -259,6 +259,53 @@
...
+				S.Target = null
+				++S.Discipline
+		if(dose == removed)
+			S.visible_message("<span class='warning'>[S]'s flesh sizzles where the water touches it!</span>", "<span class='danger'>Your flesh burns in the water!</span>")
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: This isn't water.

File: code/modules/reagents/reagent_containers.dm [hunk 17]
```diff
@@ -8,6 +8,35 @@
 	var/possible_transfer_amounts = list(5,10,15,25,30)
 	var/volume = 30
 	var/accuracy = 1
+	var/can_be_placed_into = list(
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers.dm]: This list should be indented.

File: code/modules/reagents/reagent_containers.dm [hunk 18]
```diff
@@ -8,6 +8,35 @@
...
+	/mob/living/simple_animal/hostile/retaliate/goat,
+	/obj/machinery/computer/centrifuge,
+	/obj/machinery/sleeper,
+	/obj/machinery/smartfridge,
+	/obj/machinery/biogenerator,
+	/obj/machinery/constructable_frame,
+	/obj/machinery/radiocarbon_spectrometer,
+	/obj/item/weapon/storage/part_replacer
+	)
+	//The above list a misnomer. This basically means that anything in this list has their own way of handling reagent transfers and should be ignored in afterattack.
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers.dm]: Then maybe fix the name if it's a misnomer?

File: code/modules/reagents/reagent_containers/spray.dm [hunk 19]
```diff
@@ -212,6 +215,19 @@
 			D.set_up(my_target, rand(6, 8), 2)
 	return
 
+/obj/item/weapon/reagent_containers/spray/chemsprayer/xenobiology
+	name = "xenoblaster"
+	desc = "Literally a child's plastic watergun repurposed for the use in pacifying slimes. Has an adjustable nozzle that controls precision as well as strength."
```
Comment (Lohikar) [inline code/modules/reagents/reagent_containers/spray.dm]: Kind of awkwardly phrased, primarily the use of `literally`.

File: code/modules/reagents/reagent_dispenser.dm [hunk 20]
```diff
@@ -1,72 +1,102 @@
...
 
 	var/amount_per_transfer_from_this = 10
-	var/possible_transfer_amounts = list(10,25,50,100)
+	var/possible_transfer_amounts = list(5,10,15,25,30,50,60,100,120,250,300)
 	var/capacity = 1000
-	attackby(obj/item/weapon/W as obj, mob/user as mob)
+	var/can_tamper = TRUE
+	var/is_leaking = FALSE
+
+/obj/structure/reagent_dispensers/New()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_dispenser.dm]: Initialize.

File: code/modules/reagents/reagent_dispenser.dm [hunk 21]
```diff
@@ -1,72 +1,102 @@
...
+	var/datum/reagents/R = new/datum/reagents(capacity)
```
Comment (Lohikar) [inline code/modules/reagents/reagent_dispenser.dm]: Use `create_reagents()`.

File: code/modules/reagents/reagent_dispenser.dm [hunk 22]
```diff
@@ -1,72 +1,102 @@
...
+		src.verbs -= /obj/structure/reagent_dispensers/verb/set_APTFT
+	..()
+
+/obj/structure/reagent_dispensers/examine(mob/user)
+	if(!..(user, 2))
 		return
+	user << "<span class='notice'>It contains:</span>"
+	if(reagents && reagents.reagent_list.len)
+		for(var/datum/reagent/R in reagents.reagent_list)
+			user << "<span class='notice'>[R.volume] units of [R.name]</span>"
```
Comment (Lohikar) [inline code/modules/reagents/reagent_dispenser.dm]: How can the user determine what reagents are in it from a glance?

File: code/modules/reagents/reagent_dispenser.dm [hunk 23]
```diff
@@ -75,11 +105,22 @@
 		..()
 		reagents.add_reagent("water",capacity)
 
+/obj/structure/reagent_dispensers/extinguisher
+	name = "extinguisher tank"
+	desc = "A tank filled with extinguisher fluid."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "extinguisher_tank"
+	amount_per_transfer_from_this = 10
+	New()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_dispenser.dm]: Initialize.

File: code/modules/reagents/reagent_dispenser.dm [hunk 24]
```diff
@@ -160,12 +193,10 @@
 			log_game("[key_name(Proj.firer)] shot fueltank at [loc.loc.name] ([loc.x],[loc.y],[loc.z]).",ckey=key_name(Proj.firer))
 
 		if(!istype(Proj ,/obj/item/projectile/beam/lastertag) && !istype(Proj ,/obj/item/projectile/beam/practice) )
-			explode()
+			ex_act()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_dispenser.dm]: `ex_act()` expects a param.

File: code/modules/reagents/reagent_dispenser.dm [hunk 25]
```diff
@@ -304,12 +325,4 @@
...
-		explode()
-
-/obj/structure/reagent_dispensers/cookingoil/ex_act()
-	explode()
-
-/obj/structure/reagent_dispensers/cookingoil/proc/explode()
-	reagents.splash_area(get_turf(src), 3)
-	visible_message(span("danger", "The [src] bursts open, spreading oil all over the area."))
-	qdel(src)
+		ex_act()
```
Comment (Lohikar) [inline code/modules/reagents/reagent_dispenser.dm]: Params.

File: code/modules/reagents/reagent_dispenser.dm (hunk 22 above)
Comment (BurgerLUA) [reply code/modules/reagents/reagent_dispenser.dm]: Oh, good point. Before it said "Contains x units of fuel" so I did the OBVIOUS thing, not the smart thing, and made it list all the reagents.

File: code/game/objects/effects/decals/Cleanable/fuel.dm [hunk 26]
```diff
@@ -107,4 +107,51 @@
...
+	desc = "Some kind of extinguishing foam."
+	gender = PLURAL
+	density = 0
+	anchored = 1
+	layer = 2
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "white_foam"
+	var/amount = 1
+
+	Initialize(mapload, amt = 1, nologs = 0)
```
Comment (Arrow768) [inline code/game/objects/effects/decals/Cleanable/fuel.dm]: Change that to a absolute path

File: code/game/objects/effects/decals/Cleanable/fuel.dm [hunk 27]
```diff
@@ -107,4 +107,51 @@
...
+				has_spread = 1
+				break
+
+		if(!has_spread)
+			Spread()
+			QDEL_IN(src, 2 MINUTES)
+		else
+			qdel(src)
+
+	proc/Spread(exclude=list())
```
Comment (Arrow768) [inline code/game/objects/effects/decals/Cleanable/fuel.dm]: Change that to a absolute path

File: code/game/objects/items/weapons/extinguisher.dm [hunk 28]
```diff
@@ -1,3 +1,76 @@
...
+		if(reagents && reagents.reagent_list.len)
+			to_chat(user,"<span class='notice'>The reagents are secured in the aerosol mix.</span>")
+		else
+			to_chat(user,"<span class='notice'>The cartridge seems spent.</span>")
+
+/obj/item/weapon/reagent_containers/extinguisher_refill/filled
+	name = "extinguisher refiller (monoammonium phosphate)"
+	desc = "A one time use extinguisher refiller that allows fire extinguishers to be refilled with an aerosol mix. This one is contains monoammonium phosphate."
+
+	Initialize()
```
Comment (Arrow768) [inline code/game/objects/items/weapons/extinguisher.dm]: Relative path. Change to absolute

File: code/game/turfs/simulated.dm [hunk 29]
```diff
@@ -18,26 +19,53 @@
...
-		return
+/turf/simulated/proc/wet_floor(var/apply_type = WET_TYPE_WATER, var/amount = 1)
+
+	//Wet type:
+	//WET_TYPE_WATER = water
+	//WET_TYPE_LUBE = lube
+	//WET_TYPE_ICE = ice
+
+	if(!wet_type)
+		wet_type = type
```
Comment (Arrow768) [inline code/game/turfs/simulated.dm]: Should this be
`wet_type = apply_type`

File: code/game/turfs/simulated.dm [hunk 30]
```diff
@@ -18,26 +19,53 @@
...
+	else if(type != wet_type)
```
Comment (Arrow768) [inline code/game/turfs/simulated.dm]: Shouldnt this be
`wet_type != apply_type`

File: code/game/turfs/simulated.dm [hunk 31]
```diff
@@ -18,26 +19,53 @@
...
+		if(type == WET_TYPE_WATER && wet_type == WET_TYPE_LUBE)
```
Comment (Arrow768) [inline code/game/turfs/simulated.dm]: shouldnt this be
`apply_type == WET_TYPE_WATER && wet_type == WET_TYPE_LUBE`

**Comment by Arrow768** [comment]
Review incoming. DNM for now

File: code/game/turfs/simulated.dm [hunk 32]
```diff
@@ -18,26 +19,53 @@
...
+			wet_type = WET_TYPE_WATER
+		else if(type == WET_TYPE_ICE && (wet_type == WET_TYPE_WATER || wet_type == WET_TYPE_LUBE))
```
Comment (Arrow768) [inline code/game/turfs/simulated.dm]: Shouldnt this be
`apply_type == WET_TYPE_ICE && (wet_type == WET_TYPE_WATER || wet_type == WET_TYPE_LUBE)`
or
`wet_type == WET_TYPE_ICE && (apply_type == WET_TYPE_WATER || apply_type == WET_TYPE_LUBE)`
Depending on the intention

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 33]
```diff
@@ -259,6 +259,51 @@
...
+
+/datum/reagent/toxin/fertilizer/monoammoniumphosphate/touch_turf(var/turf/simulated/T)
+
+	if(!istype(T))
+		return
+
+	var/hotspot = (locate(/obj/fire) in T)
+	if(hotspot && !istype(T, /turf/space))
+		var/datum/gas_mixture/lowertemp = T.return_air()
+		lowertemp.temperature = max(min(lowertemp.temperature-2000, lowertemp.temperature / 2), 0)
```
Comment (Arrow768) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: Isnt a guaranteed temperature reduction by 2000 °C or 50% (whichever is higher) a bit excessive ?

File: code/modules/reagents/reagent_dispenser.dm [hunk 34]
```diff
@@ -1,92 +1,126 @@
...
 	icon_state = "weldtank"
+	accept_any_reagent = FALSE
 	amount_per_transfer_from_this = 10
 	var/modded = 0
 	var/defuse = 0
 	var/armed = 0
 	var/obj/item/device/assembly_holder/rig = null
-	New()
-		..()
+	Initialize()
```
Comment (Arrow768) [inline code/modules/reagents/reagent_dispenser.dm]: Absolute paths

File: code/modules/reagents/reagent_dispenser.dm [hunk 35]
```diff
@@ -207,11 +219,11 @@
 	anchored = 1
 	density = 0
 	amount_per_transfer_from_this = 45
-	New()
-		..()
+	can_tamper = FALSE
+	Initialize()
```
Comment (Arrow768) [inline code/modules/reagents/reagent_dispenser.dm]: Absolute path

File: code/modules/reagents/reagent_dispenser.dm [hunk 36]
```diff
@@ -220,34 +232,35 @@
 	icon_state = "water_cooler"
 	possible_transfer_amounts = null
 	anchored = 1
-	New()
-		..()
-		reagents.add_reagent("water",500)
+	capacity = 500
+	can_tamper = FALSE
+	Initialize()
```
Comment (Arrow768) [inline code/modules/reagents/reagent_dispenser.dm]: Absolute path

File: code/modules/reagents/reagent_dispenser.dm [hunk 37]
```diff
@@ -220,34 +232,35 @@
...
-
 /obj/structure/reagent_dispensers/beerkeg
 	name = "beer keg"
 	desc = "A beer keg"
 	icon = 'icons/obj/objects.dmi'
 	icon_state = "beertankTEMP"
 	amount_per_transfer_from_this = 10
-	New()
-		..()
+	Initialize()
```
Comment (Arrow768) [inline code/modules/reagents/reagent_dispenser.dm]: Absolute path

File: code/modules/reagents/reagent_dispenser.dm [hunk 38]
```diff
@@ -256,8 +269,8 @@
 	icon = 'icons/obj/objects.dmi'
 	icon_state = "keg_xuizi"
 	amount_per_transfer_from_this = 10
-	New()
-		..()
+	Initialize()
```
Comment (Arrow768) [inline code/modules/reagents/reagent_dispenser.dm]: Absolute path

File: code/modules/reagents/reagent_dispenser.dm [hunk 39]
```diff
@@ -268,9 +281,9 @@
 	amount_per_transfer_from_this = 10
 	anchored = 1
 	density = 0
-
-	New()
-		..()
+	can_tamper = FALSE
+	Initialize()
```
Comment (Arrow768) [inline code/modules/reagents/reagent_dispenser.dm]: Absolute Path

File: code/modules/reagents/reagent_dispenser.dm [hunk 40]
```diff
@@ -281,12 +294,11 @@
...
```
Comment (Arrow768) [inline code/modules/reagents/reagent_dispenser.dm]: Absolute Path

File: code/modules/reagents/reagent_dispenser.dm [hunk 41]
```diff
@@ -295,19 +307,10 @@
 	icon_state = "oiltank"
 	amount_per_transfer_from_this = 120
 	capacity = 5000
-
-/obj/structure/reagent_dispensers/cookingoil/Initialize()
-	. = ..()
-	reagents.add_reagent("cornoil",capacity)
+	Initialize()
```
Comment (Arrow768) [inline code/modules/reagents/reagent_dispenser.dm]: Absolute Path

File: code/modules/research/xenoarchaeology/machinery/coolant.dm [hunk 42]
```diff
@@ -4,16 +4,16 @@
 	icon = 'icons/obj/objects.dmi'
 	icon_state = "coolanttank"
 	amount_per_transfer_from_this = 10
-	New()
-		..()
+	Initialize()
```
Comment (Arrow768) [inline code/modules/research/xenoarchaeology/machinery/coolant.dm]: Absolute Path

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm (hunk 33 above)
Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: I copied watercode.

**Comment by BurgerLUA** [comment]
If travis doesn't have any objections, it is finished @skull132

**Comment by BurgerLUA** [comment]
SPEEDMERGE GO
